repo_policy_compliance/github_client.py

```diff
@@ -129,3 +134,20 @@ def get_branch(github_client: Github, repository_name: str, branch_name: str) ->
     """
     repository = github_client.get_repo(repository_name)
     return repository.get_branch(branch_name)
+
+
+def get_collaborator_permission(
+    repository: Repository, username: str
+) -> Literal["admin", "write", "read", "none"]:
+    """Get user permission for a given repository.
+
+    Args:
+        repository: The repository to get collaborators for.
+        username: The github login to check for permission.
+
+    Returns:
+        The collaborator permission.
+    """
+    return cast(
+        Literal["admin", "write", "read", "none"], repository.get_collaborator_permission(username)
```

Comment:
Perhaps we check here that the permission is actually one of those and raise an error if it isn't?

Reply:
Added a check for valid return value before returning, thank you!